### PR TITLE
Refactor CI

### PR DESCRIPTION
**Pull request by kirill-sizov**
### Motivation and context
- [x] Add workflow that updates yarn.lock file for PRs from Snyk
- [x] Remove deprecated `set-output` for workflows (see more info [here](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/))

### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have added a description of my changes into [CHANGELOG](https://github.com/cvat-ai/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/cvat-ai/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

### CODE REVIEW

**Comment by kirill-sizov** [comment]
See how new workflow works [here](https://github.com/kirill-sizov/cvat/pull/4)

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -41,8 +41,8 @@ jobs:
           echo Default branch is ${DEFAULT_BRANCH}
           echo Workflow will try to get cache from commit: ${SHA}
 
-          echo ::set-output name=default_branch::${DEFAULT_BRANCH}
-          echo ::set-output name=sha::${SHA}
+          echo "default_branch=${DEFAULT_BRANCH}" >> $GITHUB_OUTPUT
+          echo "sha=${SHA}" >> $GITHUB_OUTPUT
```
Comment (klakhov) [inline .github/workflows/main.yml:45]: $GITHUB_OUTPUT echoed somewhere separately?

Comment (kirill-sizov) [reply .github/workflows/main.yml:45]: This is new requirement of Github: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

**Review by klakhov — APPROVED** [review]

**Pull request merged**